libsolidity/formal/SMTEncoder.cpp

```diff
@@ -304,10 +307,49 @@ void SMTEncoder::endVisit(Block const& _block)
 
 bool SMTEncoder::visit(InlineAssembly const& _inlineAsm)
 {
+	struct AssignedExternalsCollector: public yul::ASTWalker
+	{
+		using ExternalReferences = map<yul::Identifier const*, InlineAssemblyAnnotation::ExternalIdentifierInfo>;
+		AssignedExternalsCollector(InlineAssembly const& _inlineAsm): externalReferences(_inlineAsm.annotation().externalReferences)
+		{
+			this->operator()(_inlineAsm.operations());
+		}
+
+		ExternalReferences const& externalReferences;
+		set<VariableDeclaration const*> assignedVars;
+
+		using yul::ASTWalker::operator();
+		void operator()(yul::Assignment const& _assignment)
+		{
+			auto const& vars = _assignment.variableNames;
+			for (auto const& identifier: vars)
+				if (auto it = externalReferences.find(&identifier); it != externalReferences.end())
+					if (auto declaration = dynamic_cast<VariableDeclaration const*>(it->second.declaration))
+						assignedVars.insert(declaration);
+		}
+	};
+
+	auto assignedVars = AssignedExternalsCollector(_inlineAsm).assignedVars;
```

Comment:
I wonder if we really have to write the class `AssignedExternalCollector`.

There is `Assignments` in `libyul/optimiser/NameCollector.h`, which should give you all `_assignment.variableNames`. Then you could just iterate through the names, without having to write a class. What do you think?

Reply:
`NameCollector` seems to be collecting `Yul` variables declared in the assembly, at least that's how I am reading it.
What I need are the Solidity local variables referenced (and more precisely assigned) in the assembly.
So I think `NameCollector` is not the right tool, but I will check if there is not something else, thanks for the tip!

Reply:
Sorry, I misread your comment initially,`Assignments` looks useful, thanks for the tip!

Reply:
`Assignments` would only give `YulString`. If you really need `Identifier`, then you might have to write a walker class. So not sure.

Reply:
Yeah, `Identifier`s would be much better, since I am then accessing a map with `Identifier *` type of keys. I could work around that, but collecting directly `Identifier`s seems cleaner to me.